master/internal/usergroup/api_groups.go

```diff
@@ -95,7 +120,18 @@ func (a *UserGroupAPIServer) GetGroup(ctx context.Context, req *apiv1.GetGroupRe
 		err = apiutils.MapAndFilterErrors(err)
 	}()
 
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
 	gid := int(req.GroupId)
+
+	err = AuthZProvider.Get().CanGetGroup(*curUser, gid)
```

Comment:
I think we want ```CanGetGroup``` to return the boolean for if you can see the group and the server error 

```
if canSee, serverErr = AuthZProvider.Get().CanGetGroup(*curUser, gid); serverErr != nil {
    return serverError
} else if !canSee {
    return errors.Wrapf(db.ErrNotFound, "Error getting group %d", gid) // Same error as when group actually doesn't 
}
```